app/src/main/kotlin/com/simplemobiletools/keyboard/extensions/Context.kt

```diff
@@ -19,17 +19,21 @@ import com.simplemobiletools.keyboard.databases.ClipsDatabase
 import com.simplemobiletools.keyboard.helpers.*
 import com.simplemobiletools.keyboard.interfaces.ClipsDao
 
+
 val Context.config: Config get() = Config.newInstance(applicationContext.safeStorageContext)
 
 val Context.safeStorageContext: Context
-    get() = if (isNougatPlus() && isDeviceLocked) {
+    get() = if (isNougatPlus() && isDeviceInDirectBootMode) {
         createDeviceProtectedStorageContext()
     } else {
         this
     }
 
-val Context.isDeviceLocked: Boolean
-    get() = (getSystemService(Context.KEYGUARD_SERVICE) as KeyguardManager).isDeviceLocked
+val Context.isDeviceInDirectBootMode: Boolean
+    get() {
+        val userManager = getSystemService(Context.USER_SERVICE) as UserManager
+        return isNougatPlus() && userManager.isUserUnlocked
```

Comment:
shouldnt this be !userManager.isUserUnlocked ?

Reply:
Certainly! How it managed to work like that on my phone is a mystery